### PR TITLE
feat: udev: label device nodes

### PR DESCRIPTION
**Pull request by dsseng**
Use udev rules to assign basic device file labels based on their subsystem

Signed-off-by: Dmitry Sharshakov <dmitry.sharshakov@siderolabs.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -731,6 +731,7 @@ COPY --chmod=0644 hack/containerd.toml /rootfs/etc/containerd/config.toml
 COPY --chmod=0644 hack/cri-containerd.toml /rootfs/etc/cri/containerd.toml
 COPY --chmod=0644 hack/cri-plugin.part /rootfs/etc/cri/conf.d/00-base.part
 COPY --chmod=0644 hack/udevd/80-net-name-slot.rules /rootfs/usr/lib/udev/rules.d/
+COPY --chmod=0644 hack/udevd/xx-selinux.rules /rootfs/usr/lib/udev/rules.d/
```
Comment (dsseng) [inline Dockerfile:734]: In what place in order should we place it? Perhaps 99?

File: internal/pkg/selinux/policy/selinux/services/udev.cil [hunk 2]
```diff
@@ -20,6 +20,20 @@
...
+; RTC is important for security verification
+(type rtc_device_t)
+(call protected_device_f (rtc_device_t))
+; Could be exposed firmware storage
+(type mtd_device_t)
+(call protected_device_f (mtd_device_t))
+; Could reset the system
+(type wdt_device_t)
+(call protected_device_f (wdt_device_t))
+; TODO: should pods be able to access TPM?
```
Comment (dsseng) [inline internal/pkg/selinux/policy/selinux/services/udev.cil:33]: Should they? If so, no reason to label TPM specially if we are not going to use its label.

File: Dockerfile (hunk 1 above)
Comment (dsseng) [reply Dockerfile:734]: Set 90 now

File: hack/udevd/90-selinux.rules [hunk 3]
```diff
@@ -0,0 +1,11 @@
+SUBSYSTEM=="*",SECLABEL{selinux}="system_u:object_r:device_t:s0"
```
Comment (dsseng) [inline hack/udevd/90-selinux.rules:1]: Label all devices as device_t, overrides are lower in this file

File: Dockerfile (hunk 1 above)
Comment (frezbo) [reply Dockerfile]: 90 feels good, maybe we should also do a validation that the user doesn't override these?

File: hack/udevd/90-selinux.rules (hunk 3 above)
Comment (frezbo) [inline hack/udevd/90-selinux.rules:1]: are these labels like standard? I guess devices can have standard labels

File: internal/integration/api/selinux.go [hunk 4]
```diff
@@ -117,12 +117,25 @@ func (suite *SELinuxSuite) TestFileMountLabels() {
 	}
 	maps.Copy(expectedLabelsControlPlane, expectedLabelsWorker)
 
-	suite.checkFileLabels(workers, expectedLabelsWorker)
-	suite.checkFileLabels(controlplanes, expectedLabelsControlPlane)
+	// Devices labeled by subsystems, labeled by udev
+	expectedLabelsDevices := map[string]string{
+		"/dev/rtc0":      "system_u:object_r:rtc_device_t:s0",
+		"/dev/tpm0":      "system_u:object_r:tpm_device_t:s0",
```
Comment (frezbo) [inline internal/integration/api/selinux.go:123]: hmm `/dev/tpm` is not available by default on our qemu tests, right? only for secureboot i believe

Comment (dsseng) [reply internal/integration/api/selinux.go:123]: Yes. I added an option for test to ignore when file doesn't exist

File: hack/udevd/90-selinux.rules (hunk 3 above)
Comment (dsseng) [reply hack/udevd/90-selinux.rules:1]: Apparently `(fsuse trans devtmpfs (system_u object_r device_t (systemLow systemLow)))` is not sufficient for this to be default

File: Dockerfile (hunk 1 above)
Comment (dsseng) [reply Dockerfile]: Well, user is the Talos admin, not the k8s admin who we protect the system from. Perhaps should be okay

Comment (frezbo) [reply Dockerfile]: I guess it's fine, on second thoughts I believe we allow overwriting existing udev rules, so should be okay

File: internal/integration/api/selinux.go (hunk 4 above)
Comment (frezbo) [reply internal/integration/api/selinux.go:125]: actually let's just enable tpm2 for all tests in CI, it shouldn't hurt, better than trying to match on error, probably can do in a different PR

**Review by frezbo — APPROVED** [review]

File: internal/integration/api/selinux.go (hunk 4 above)
Comment (dsseng) [reply internal/integration/api/selinux.go:125]: Will do

**Pull request merged**